### PR TITLE
Allow long (integration) tests for go

### PR DESCRIPTION
**Pull request by dorner**
Tested with PD-267 branch on distribution-api: https://github.com/wishabi/distribution-api/actions/runs/5248453591/jobs/9479970029

### CODE REVIEW

File: go/test/action.yml [hunk 1]
```diff
@@ -1,12 +1,21 @@
 name: 'go-test'
 description: 'Custom GitHub action that runs all Golang tests within a repository and saves the test + coverage reports as artifacts'
 
+inputs:
+  SHORT:
+    description: Set to true if you want to run all tests, including long running tests.
+    required: false
```
Comment (chriskarlin) [inline go/test/action.yml:7]: Shouldn't this be flipped? When `short` is set to `true` it should SKIP long running tests

Comment (dorner) [reply go/test/action.yml:7]: lol yes! But this actually mirrors the `-short` argument for `go test`. I have no idea why they did it this way.

Comment (chriskarlin) [reply go/test/action.yml:7]: Does it? https://pkg.go.dev/testing#hdr-Skipping
This reads like I understood it, if set (true) skip tests

Comment (chriskarlin) [reply go/test/action.yml:7]: https://pkg.go.dev/testing#Short

Comment (dorner) [reply go/test/action.yml:7]: `short` is just a flag that returns true. You can use it either to add or skip tests. Practically speaking there will be fewer integration (long) tests than short ones, so it seems like it's more common to use this to mark integration tests and skip them if the flag isn't turned on.

**Review by chriskarlin — APPROVED** [review]

File: go/test/action.yml (hunk 1 above)
Comment (dorner) [reply go/test/action.yml:7]: ...nope, I completely misunderstood this. [https://blog.dharnitski.com/2019/04/29/skip-tests-in-go/](https://blog.dharnitski.com/2019/04/29/skip-tests-in-go/) This seems really weird. Let me have another go at this.

Comment (dorner) [reply go/test/action.yml:7]: Maybe using tags and build comments is a better approach.

**Comment by dorner** [comment]
Going to open a new PR, commit history got borked.

**Pull request closed without merging**